app/src/main/java/com/amaze/filemanager/ui/activities/FlashScreen.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2014-2022 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
+ * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
+ *
+ * This file is part of Amaze File Manager.
+ *
+ * Amaze File Manager is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.ui.activities
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.animation.AnimationUtils
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.amaze.filemanager.R
+import java.util.*
+
+class FlashScreen : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_flash_screen)
+        val text1 = findViewById<TextView>(R.id.flash_text)
+        text1.startAnimation(AnimationUtils.loadAnimation(this, android.R.anim.slide_in_left))
+        Timer().schedule(
+            object : TimerTask() {
+                override fun run() {
+                    startActivity(Intent(this@FlashScreen, MainActivity::class.java))
```

Comment:
The activity should be started before the timer starts, so that it is available as fast as possible.

Reply:
actually i am not able to figure how can i do that , can you give me a hint for that

Reply:
I think you can just do `startActivity()` before the timer task. That's how I did it for a game some time ago:

https://github.com/EmmanuelMess/Tactical-Defence/blob/0af05b4767e8f2ce1e9b326a709571fd4f09e9d1/TacticalDefence/app/src/main/java/com/emmanuelmess/tacticaldefence/activities/game/GameMenuActivity.java#L140-L143

Reply:
Okay, I have tried it and it doesn't work (anymore?) you'll probably need to do another thing.

Reply:
You'll probably have to reverse engineer the API 16 activity to check how it does it, and make it backwards compatible. Using `onWindowFocusChanged(hasFocus: Boolean)` works for showing the splash but not for loading the new activity.

Remember a requirement is that the MainAcitivity starts loading immediately (within a few ms) after the splash activity starts.